lib/view.js

```diff
@@ -0,0 +1,230 @@
+/** @babel */
+
+import {Emitter} from 'event-kit';
+
+import {Path} from './models';
+import {cachedProperty, closest, dom} from './utils';
+
+
+export default class AdvancedOpenFileView {
+    constructor() {
+        this.emitter = new Emitter();
+        this.cursorIndex = null;
+        this._updatingPath = false;
+
+        // Element references
+        this.pathInput = this.content.querySelector('.path-input');
+        this.pathList = this.content.querySelector('.list-group');
+        this.parentDirectoryListItem = this.content.querySelector('.parent-directory');
```

Comment:
I previously did have a separate event handler for clicks on the parent element and the controller was the one that handled clicks to it. I ended up not being a fan of the extra code required in the controller vs. just making it a magical special element with the parent path.

The alternative to having a special handler for it would be to have the controller and model deal with path lists that have the parent directory in them, which might work. But I don't see a whole lot of value in that change and I'm not married to the idea of being 100% strict MVC anyway. But your point is still valid, it is an awkward thing. Worth mulling over if I get bored.